### PR TITLE
(feature) Add resursive resolve for allOf.

### PR DESCRIPTION
**Pull request by dumconstantin**

### CODE REVIEW

**Comment by loganvolkers** [comment]
Thanks for the contribution @dumconstantin! I'll take a look right away.

**Comment by mattiasrunge** [comment]
Is there any plan to merge this pull-request and do a new release on npm?
This is currently a blocker in a project I am working on.
Thanks for this useful library!

**Comment by loganvolkers** [comment]
@mattiasrunge thanks for checking in! 🥇 👍 

I have been keeping an eye on this. Waiting for a response from @dumconstantin to my comments. 🕐

**Comment by forrest-ua** [comment]
@loganvolkers is there anything we can help with meanwhile in case @dumconstantin will not reply?

**Comment by dumconstantin** [comment]
Hi guys,

@loganvolkers I didn't see any comments, but I see there is a conflict with the index.* files which I added by mistake (sorry bout that!).

Is there anything else besides those files?

File: index.ts [hunk 1]
```diff
@@ -8,7 +8,10 @@ module.exports = function resolveAllOf(inputSpec){
             var allOf = inputSpec.allOf;
             delete inputSpec.allOf;
             var nested = _.mergeWith({}, ...allOf, customizer);
-            out = _.defaults(inputSpec, nested, customizer);
+            out = _.defaultsDeep(inputSpec, nested, customizer);
```
[the remark this answers is not in the document]
Comment (loganvolkers) [reply index.ts]: Ignore this -- I just checked it out locally and ran tests, and I can see how a deep merge is required. Thanks for the test case!

**Comment by loganvolkers** [comment]
@dumconstantin just the conflicts left to resolve

**Comment by dumconstantin** [comment]
I tried updating with your master branch but I can't get the tests to succesfully run.

I keep getting:
```
ERROR: index.ts[5, 9]: Merge this if statement with the enclosing one.
```
which I guess refers to:
```
    if(inputSpec && typeof inputSpec === 'object'){
        if(inputSpec.allOf){ ... }
```
but that doesn't make sense to merge due to the code that follows... so I guess the ` resolveAllOf ` function needs a refactor to comply to Typescript's rules?

**Comment by loganvolkers** [comment]
Closed in #5

**Pull request closed without merging**